src/popups/PopupContextMenu.as

```diff
@@ -162,14 +163,15 @@ package popups
             {
                 MultiplayerSingleton.destroyInstance();
                 Flags.VALUES = {};
+                Playlist.clearCanon();
                 _gvars.gameMain.switchTo("none");
             }
             else if (e.target.action == "switch_profile")
             {
                 MultiplayerSingleton.destroyInstance();
                 Flags.VALUES = {};
                 _gvars.playerUser.refreshUser();
-                _gvars.gameMain.switchTo("GameLoginPanel");
+                _gvars.gameMain.switchTo(Main.GAME_LOGIN_PANEL);
```

Comment:
Same here, an `if (_gvars.gameMain.preloader.isComplete)` block should surround this.